.circleci/config.yml

```diff
@@ -43,19 +44,52 @@ jobs:
            echo "export START_TIME=$(date +%Y%m%d-%Hh%Mm%Ss)" >>$BASH_ENV
            echo "export GCP_IMAGE=gcr.io/${GOOGLE_PROJECT_ID}/${IMAGE}" >> $BASH_ENV
            echo "export PR=$( [[ -z ${CIRCLE_PULL_REQUEST+x} ]] && echo false || echo $(basename ${CIRCLE_PULL_REQUEST}) )" >>$BASH_ENV
-           #echo "export HELM_TEST_NAME=$(echo $CIRCLE_BRANCH | sed 's/[^A-Za-z0-9\\-]*//g' | tr '[:upper:]' '[:lower:]')" >>$BASH_ENV
            export PR=$( [[ -z ${CIRCLE_PULL_REQUEST+x} ]] && echo false || echo $(basename ${CIRCLE_PULL_REQUEST}) )           
            echo "export HELM_TEST_NAME=pr-${PR}" >> $BASH_ENV
            sudo service apport stop
+
       - run:
-         name: Docker Build taraxa-node
-         command: |
-           docker build -t ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} .
-      - run:
-         name: Docker Build taraxa-node-ctest
+         name: Login into gcr
          command: |
-          docker build -t ${IMAGE}-ctest:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} \
-            --target build .
+            echo ${GCLOUD_SERVICE_KEY} | docker login -u _json_key --password-stdin https://gcr.io
+ 
+#      - run:
+#         name: Docker Build taraxa-node
+#         command: |
+#           docker build -t ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} .
+
+      - docker-cache/with-save-restore-images:
+         repository: gcr.io/${GOOGLE_PROJECT_ID}
+         images: >-
+           ${IMAGE}:${DOCKER_BRANCH_TAG}
+           ${IMAGE}-cache:${DOCKER_BRANCH_TAG}
+           ${IMAGE}-cache:latest
```

Comment:
Why do we need this cache image ?

Reply:
If we only save the actual cache/ctest image there, I think it's a good idea, so we can from time to time, clean up the registry and save some storage costs.

Reply:
the Removed ${IMAGE}:${DOCKER_BRANCH_TAG} but the other two are required, we may choose not to use, the latest, but as we wred discussing, the latest is inteneded for the 'first commit in pr' while the one with the branch should be useful for the next runs, they should share more layers but when downloading they differ more than i expected.